app/models/trip_dashboard.rb

```diff
@@ -0,0 +1,97 @@
+module TripDashboard
+
+  def average_duration_of_ride
+    return 0 if count.zero?
+    (average("duration").round(1) / 60).round(2)
+  end
+
+  def longest_ride
+    return 0 if count.zero?
+    (maximum("duration").round(1) / 60).round(2)
+  end
+
+  def shortest_ride
+    return 0 if count.zero?
+    (minimum("duration").round(1) / 60).round(2)
+  end
+
+  def start_station_with_most_rides
+    return "" if count.zero?
+    station = Station.joins(:trips).group(:name).count("id")
+    station.keys.max_by {|key| station[key]}
+  end
+
+  def end_station_with_most_rides
+    return "" if count.zero?
+    trips = Trip.joins(:station).group(:end_station_id).count("id")
+    station_id = trips.keys.max_by {|key|trips[key]}
+    Station.find(station_id).name
+  end
+
+  def rides_per_month(mon, yea)
+    return 0 if count.zero?
+    month_trips = Trip.where('extract(month FROM start_date) = ?',mon)
+    year_trips = month_trips.where('extract(year FROM start_date)= ?', yea)
+    year_trips.count
+  end
+
+  def best_bike
+    bikes = Trip.group(:bike_id).count("id")
+    bikes.keys.max_by do |key|
+      bikes[key]
+    end
+  end
+
+  def best_bike_count
+    bikes = Trip.group(:bike_id).count("id")
+    bikes[best_bike]
+  end
+
+  def worst_bike
+    bikes = Trip.group(:bike_id).count("id")
+    bikes.keys.min_by {|key|bikes[key]}
+  end
+
+  def worst_bike_count
+    bikes = Trip.group(:bike_id).count("id")
+    bikes[worst_bike]
+  end
+
+  def sub_type_count
+    subs = SubscriptionType.joins(:trips).group(:name).count("id")
+    subs.values
+  end
+
+  def sub_percent
+    total = SubscriptionType.joins(:trips).group(:name).count("id").values.sum
+    ((sub_type_count[1].to_f / total.to_f) * 100).round(2)
+  end
+
+  def cust_percent
+    total = SubscriptionType.joins(:trips).group(:name).count("id").values.sum
```

Comment:
we're also calling this multiple times - can we extract it to it's own method and just call it rather than re-writing it?